### PR TITLE
T11: add signin page functionality 

### PR DESCRIPTION
**Pull request by idaosk**
created a viewmodel & activity for sign in

### CODE REVIEW

**Comment by idaosk** [comment]
Ii created the ViewModel class to the best of my knowledge but I need a tutorial on this subject to learn properly. 
@moyarsuccess @amirziarati

File: app/build.gradle [hunk 1]
```diff
@@ -50,7 +51,16 @@ dependencies {
     androidTestImplementation 'androidx.test.ext:junit:1.1.5'
     androidTestImplementation 'androidx.test.espresso:espresso-core:3.5.1'
 
+    //Firebase Authentication
+    implementation 'com.google.firebase:firebase-auth-ktx'
+    //Firebase Analytics
+    implementation 'com.google.firebase:firebase-analytics-ktx'
```
[the remark this answers is not in the document]
Comment (Animansoubi) [reply app/build.gradle]: yes, I've already added them.

**Review by Animansoubi — COMMENTED** [review]
LGTM

File: app/src/main/java/com/gathering/android/Application.kt [hunk 2]
```diff
@@ -1,7 +1,10 @@
 package com.gathering.android
 
 import android.app.Application
+import com.google.firebase.FirebaseApp
```
Comment (moyarsuccess) [inline app/src/main/java/com/gathering/android/Application.kt:4]: This dependency here is extra.

File: app/src/main/java/com/gathering/android/Application.kt [hunk 3]
```diff
@@ -1,7 +1,10 @@
...
 import dagger.hilt.android.HiltAndroidApp
 
 @HiltAndroidApp
-class Application : Application()
\ No newline at end of file
+class Application : Application() {
+}
```
Comment (moyarsuccess) [inline app/src/main/java/com/gathering/android/Application.kt:9]: Why you have added these curly braces?

**Review by moyarsuccess — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/gathering/android/Application.kt (hunk 2 above)
Comment (idaosk) [reply app/src/main/java/com/gathering/android/Application.kt]: done!

**Review by moyarsuccess — APPROVED** [review]

**Review by moyarsuccess — APPROVED** [review]

**Pull request merged**